### ISSUE TITLE
GenerateAvroJavaTask stuck in an infinite loop in case duplicate type names in the same file

### ISSUE THREAD

**Issue by shardulm94** [commercehub-oss/gradle-avro-plugin#47]
Ideally, the task should throw an exception in such a case and not get stuck in an infinite loop
Example avsc file where duplicate type name `date` exists.
```
{
    "type" : "record",
    "name" : "Fail",
    "namespace" : "example.avro",
    "fields" : [ {
        "name" : "start_date",
        "type" : {
            "type" : "record",
            "name" : "date",
            "fields" : [ {
                "name" : "day",
                "type" : ["null", "int"]
            }, {
                "name" : "month",
                "type" : ["null", "int"]
            }, {
                "name" : "year",
                "type" : ["null", "int"]
            } ]
        }
    }, {
        "name" : "end_date",
        "type" : {
            "type" : "record",
            "name" : "date",
            "fields" : [ {
                "name" : "day",
                "type" : ["null", "int"]
            }, {
                "name" : "month",
                "type" : ["null", "int"]
            }, {
                "name" : "year",
                "type" : ["null", "int"]
            } ]
        }
    } ]
}
```

Debug Log:
```
01:34:14.833 [DEBUG] [org.gradle.api.Task] Processing custom.avsc, excluding types [example.avro.date]
01:34:14.833 [DEBUG] [org.gradle.api.Task] Identified duplicate type example.avro.date in custom.avsc; will re-process excluding it
01:34:14.833 [DEBUG] [org.gradle.api.Task] Processing custom.avsc, excluding types [example.avro.date]
01:34:14.833 [DEBUG] [org.gradle.api.Task] Identified duplicate type example.avro.date in custom.avsc; will re-process excluding it
01:34:14.833 [DEBUG] [org.gradle.api.Task] Processing custom.avsc, excluding types [example.avro.date]
01:34:14.833 [DEBUG] [org.gradle.api.Task] Identified duplicate type example.avro.date in custom.avsc; will re-process excluding it
01:34:14.833 [DEBUG] [org.gradle.api.Task] Processing custom.avsc, excluding types [example.avro.date]
01:34:14.834 [DEBUG] [org.gradle.api.Task] Identified duplicate type example.avro.date in custom.avsc; will re-process excluding it
01:34:14.834 [DEBUG] [org.gradle.api.Task] Processing custom.avsc, excluding types [example.avro.date]
01:34:14.834 [DEBUG] [org.gradle.api.Task] Identified duplicate type example.avro.date in custom.avsc; will re-process excluding it
01:34:14.834 [DEBUG] [org.gradle.api.Task] Processing custom.avsc, excluding types [example.avro.date]
01:34:14.834 [DEBUG] [org.gradle.api.Task] Identified duplicate type example.avro.date in custom.avsc; will re-process excluding it
01:34:14.834 [DEBUG] [org.gradle.api.Task] Processing custom.avsc, excluding types [example.avro.date]
01:34:14.834 [DEBUG] [org.gradle.api.Task] Identified duplicate type example.avro.date in custom.avsc; will re-process excluding it
```

**Comment by davidmc24** [comment]
Thanks for the bug report. I should be able to investigate tomorrow.

**Comment by davidmc24** [comment]
Fix released in version 0.14.1.  Thanks again!
